File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        mindiff=float('inf')
        res = []
        for i in range(len(arr)):
            for j in range(i+1,len(arr)):
                mindiff = min(mindiff,abs(arr[i]-arr[j]))
        arr.sort()
        for i in range(len(arr)):
            for j in range(i+1,len(arr)):
                if abs(arr[i]-arr[j]) == mindiff:
                    res.append([arr[i],arr[j]])
        return res

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        int n = arr.length;
        List<List<Integer>> res = new ArrayList<>();
        int mindiff = Integer.MAX_VALUE;
        Arrays.sort(arr);
        for (int i=1; i<n; i++){
            mindiff = Math.min(mindiff,arr[i]-arr[i-1]);
        }
        for (int i=1; i<n; i++){
           if(arr[i]-arr[i-1] == mindiff){
                res.add(Arrays.asList(arr[i-1],arr[i]));
           }
        }
        return res;
    }
}